grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -2813,4 +2815,33 @@ class StockMovementService {
         }
         return true
     }
+
+    void rollbackApproval(String stockMovementId) {
+        StockMovement stockMovement = getStockMovement(stockMovementId)
+        if (!canRollbackApproval(AuthService.currentUser.get(), stockMovement)) {
```

Comment:
we NEED to remember to change it in Grails 3 - in Grails 3 we should use `AuthService.currentUser`, since adding `get()` makes it returning `null`.

Reply:
please add an appropiate TODO note or check if in Grails 1 it would work without `get()`

Reply:
the current user is wrapped in `ThreadLocal`
![image](https://github.com/openboxes/openboxes/assets/83239466/da3e0ee2-3204-49ec-bdb9-3311e6679205)
and this type has method get to return the stored value

Reply:
@alannadolny yes, this is the correct way to retrieve the User in Grails 1 because current user is a ThreadLocal. In Grails 3, the user object is still a threadlocal<user> but I think we added a way to retrieve the user object from the AuthService without the additional get(). In other words I think we renamed the property and added a method getCurrentUser() so that we could just invoke AuthService.currentUser to get access to the User. So @kchelstowski just wants a reminder TODO to make sure don't forget to change it when the code is ported.

Reply:
AuthService.currentUser is pointing to that line:
```
    static ThreadLocal<User> currentUser = new ThreadLocal<User>()
```
I added a comment with reminder to change that